platform/ext/target/nordic_nrf/common/plat_test.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tfm_plat_test.h"
+#include <stdint.h>
+#include <stdbool.h>
+#include <hal/nrf_gpio.h>
+#include <hal/nrf_timer.h>
+#include <nrf_board.h>
+
+#define TIMER_RELOAD_VALUE (16*1024*1024)
+#define USERLED_MASK       (1UL)
+
+static bool initialized = false;
+
+static void gpio_init(void)
+{
+    nrf_gpio_cfg_input(BUTTON1_PIN, BUTTON1_PULL);
+    nrf_gpio_cfg_output(LED1_PIN);
+    initialized = true;
+}
+
+void tfm_plat_test_wait_user_button_pressed(void)
+{
+    if (!initialized) gpio_init();
+
+    /* Wait until button is pressed */
+    while (nrf_gpio_pin_read(BUTTON1_PIN) != BUTTON1_ACTIVE_HIGH) { ; }
+}
+
+void tfm_plat_test_wait_user_button_released(void)
+{
+    if (!initialized) gpio_init();
+
+    /* Wait until user button 0 is released */
+    while (nrf_gpio_pin_read(BUTTON1_PIN) == BUTTON1_ACTIVE_HIGH) { ; }
+}
+
+uint32_t tfm_plat_test_get_led_status(void)
+{
+    if (!initialized) gpio_init();
+
+    return (nrf_gpio_pin_out_read(LED1_PIN) == LED1_ACTIVE_HIGH);
+}
+
+void tfm_plat_test_set_led_status(uint32_t status)
+{
+    if (!initialized) gpio_init();
+
+    nrf_gpio_pin_write(LED1_PIN, status == LED1_ACTIVE_HIGH);
```

Comment:
I'm afraid this won't work like that. See how the `tfm_plat_test_set_led_status()` function is used:
https://github.com/NordicPlayground/trusted-firmware-m/blob/1a02aa34328394331391284dc94dfc4e6f286b8a/test/test_services/tfm_core_test/tfm_ss_core_test.c#L209-L214
You'll need to involve `USERLED_MASK` here.

BTW Perhaps it would be better to rename the macro to `LED1_ACTIVE_LEVEL`?

Reply:
I think it should work since I've defined userled_mask to be 1. Well, maybe I should mask the status for good measure. Will do that.

Reply:
The thing is that the code I pointed out does not use `userled_mask`, though it probably should.

Reply:
Yep, I was thinking
```suggestion
    status &= USERLED_MASK;
    nrf_gpio_pin_write(LED1_PIN, status == LED1_ACTIVE_HIGH);
```